### ISSUE TITLE
RCS1031: False positive when local variables share names

### ISSUE THREAD

**Issue by eduherminio** [JosefPihrt/Roslynator#801]
**Product and Version Used**: 

Roslynator 2019 VS Extension, v3.2.0

**Steps to Reproduce**:
```csharp
var key = "key_1";
var value = "false";

switch (key)
{
    case "key_1":
    {
        bool.TryParse(value, out var val);
        break;
    }
    case "key_2":
    {
        bool.TryParse(value, out var val);
        break;
    }
    // ...
}
```

**Actual Behavior**:
`RCS1031: Remove unnecessary braces`.
Applying the suggestion raises CS0136.

**Expected Behavior**:
No warning raised, since the braces cannot be removed without producing the compiler error due to `out var val` used in both switch sections.

**Comment by JosefPihrt** [comment]
fixed with https://github.com/JosefPihrt/Roslynator/pull/1039
